### PR TITLE
Remove (migrated) DLPack docs from Data Interchange page

### PR DESCRIPTION
**Pull request by tirthasheshpatel**
DLPack now has official docs published at: https://dmlc.github.io/dlpack/latest/

dmlc/dlpack#93 migrated the docs added in #106 under the [Python Specification of DLPack](https://dmlc.github.io/dlpack/latest/python_spec.html) page. So, these aren't needed anymore. This PR removed the migrated docs and redirects to DLPack's official doc pages.

### CODE REVIEW

File: spec/design_topics/data_interchange.rst [hunk 1]
```diff
@@ -70,116 +70,15 @@ support libraries that already implement buffer protocol support.
...
-
-Implementation
-~~~~~~~~~~~~~~
-
-*Note that while this API standard largely tries to avoid discussing
-implementation details, some discussion and requirements are needed
-here because data interchange requires coordination between
-implementers on, e.g., memory management.*
-
-.. image:: /_static/images/DLPack_diagram.png
```
Comment (leofang) [inline spec/design_topics/data_interchange.rst:137]: I think we can also remove this asset.

File: spec/design_topics/data_interchange.rst [hunk 2]
```diff
@@ -70,116 +70,15 @@ support libraries that already implement buffer protocol support.
    See the `RFC to adopt DLPack <https://github.com/data-apis/consortium-feedback/issues/1>`_
    for discussion that preceded the adoption of DLPack.
 
+DLPack's documentation can be found at:
+`https://dmlc.github.io/dlpack/latest/ <https://dmlc.github.io/dlpack/latest/>`__.
```
Comment (leofang) [inline spec/design_topics/data_interchange.rst:74]: nit: I thought a plain url would also work with Sphinx?
```suggestion
https://dmlc.github.io/dlpack/latest/.
```

File: spec/design_topics/data_interchange.rst (hunk 1 above)
Comment (tirthasheshpatel) [reply spec/design_topics/data_interchange.rst:137]: Done.

File: spec/design_topics/data_interchange.rst (hunk 2 above)
Comment (tirthasheshpatel) [reply spec/design_topics/data_interchange.rst]: Done. Thanks!

File: spec/design_topics/data_interchange.rst [hunk 3]
```diff
@@ -40,7 +40,7 @@ The interchange mechanism must offer the following:
    ABI is required for packages to work well together.*
 
 The best candidate for this protocol is
```
Comment (tirthasheshpatel) [inline spec/design_topics/data_interchange.rst:42]:
```suggestion

DLPack Support
--------------

The best candidate for this protocol is
```

@leofang How about we add the section up here? It highlights that DLPack is the chosen standard. Maybe, we can change the name to something like `Choosing DLPack` or `DLPack: An in-memory tensor structure`

Comment (leofang) [reply spec/design_topics/data_interchange.rst:42]: Yup that'd work too!

**Review by leofang — APPROVED** [review]

**Review by kgryte — APPROVED** [review]
LGTM

**Comment by kgryte** [comment]
Thanks for reviewing @leofang!

**Review by rgommers — APPROVED** [review]
LGTM too. Thanks @tirthasheshpatel & reviewers!

**Pull request merged**